### PR TITLE
[MultiControlNet] Allow save and load

### PR DESCRIPTION
**Pull request by patrickvonplaten**
This PR allows **local** saving and loading of multiple controlnets. The following code snippet now works:

```py
#!/usr/bin/env python3
import diffusers
import torch
import os

hugging_face_token = "<insert hugging face token>"
controlnet_dir = f"./edit-anything-v0-4-sd15/"
controlnet_inpaint_dir = f"./control_v11p_sd15_inpaint/"
sd_dir = f"./stable-diffusion-v1-5/"

edit_anything_controlnet = diffusers.ControlNetModel.from_pretrained(
    "shgao/edit-anything-v0-4-sd15",
    cache_dir=controlnet_dir,
    torch_dtype=torch.float16,
)
edit_anything_controlnet.save_pretrained(controlnet_dir, safe_serialization=True)

inpaint_controlnet = diffusers.ControlNetModel.from_pretrained(
    "lllyasviel/control_v11p_sd15_inpaint",
    cache_dir=controlnet_inpaint_dir,
    torch_dtype=torch.float16,
)

inpaint_controlnet.save_pretrained(controlnet_inpaint_dir, safe_serialization=True)

pipe = diffusers.StableDiffusionControlNetPipeline.from_pretrained(
    "runwayml/stable-diffusion-v1-5",
    controlnet=[edit_anything_controlnet, inpaint_controlnet], # this turns the controlnet into MultiControlNet
    cache_dir=sd_dir,
    torch_dtype=torch.float16,
    use_auth_token=hugging_face_token,
)
pipe.scheduler = diffusers.UniPCMultistepScheduler.from_config(pipe.scheduler.config)
pipe.save_pretrained(sd_dir, safe_serialization=True)

pipe = diffusers.StableDiffusionControlNetPipeline.from_pretrained(sd_dir)
```

This should fix the following issue: https://github.com/huggingface/diffusers/issues/3683

cc @DJBen let me know if this works for you

### CODE REVIEW

File: src/diffusers/pipelines/controlnet/multicontrolnet.py [hunk 1]
```diff
@@ -64,3 +69,117 @@ def forward(
...
+                TPUs and need to call this function on all processes. In this case, set `is_main_process=True` only on
+                the main process to avoid race conditions.
+            save_function (`Callable`):
+                The function to use to save the state dictionary. Useful on distributed training like TPUs when one
+                need to replace `torch.save` by another method. Can be configured with the environment variable
+                `DIFFUSERS_SAVE_MODE`.
+            safe_serialization (`bool`, *optional*, defaults to `False`):
+                Whether to save the model using `safetensors` or the traditional PyTorch way (that uses `pickle`).
+            variant (`str`, *optional*):
+                If specified, weights are saved in the format pytorch_model.<variant>.bin.
```
Comment (sayakpaul) [inline src/diffusers/pipelines/controlnet/multicontrolnet.py:99]: Maybe an example of what a variant should be?

File: src/diffusers/pipelines/controlnet/multicontrolnet.py [hunk 2]
```diff
@@ -64,3 +69,117 @@ def forward(
...
+
+            idx += 1
+            model_path_to_load = pretrained_model_path + f"_{idx}"
+
+        logger.info(f"{len(controlnets)} controlnets loaded from {pretrained_model_path}.")
+
+        if len(controlnets) == 0:
+            raise ValueError(
+                "No ControlNets found under {os.path.dirname(pretrained_model_path)}. Expected at least {pretrained_model_path + '_0'}."
+            )
```
Comment (sayakpaul) [inline src/diffusers/pipelines/controlnet/multicontrolnet.py:183]: `f""` missing.

**Review by sayakpaul — APPROVED** [review]
Nice change! LGTM.

File: src/diffusers/pipelines/controlnet/multicontrolnet.py (hunk 2 above)
Comment (patrickvonplaten) [reply src/diffusers/pipelines/controlnet/multicontrolnet.py:183]: Good catch!

**Comment by DJBen** [comment]
Thank you for the change!

**Pull request merged**